### DISCUSSION TITLE
Persistent LiveUSB for SpiralLinux?

### DISCUSSION THREAD

**Discussion by ping-wu** [SpiralLinux/SpiralLinux-project#8] [Q&A] [answered]
Nowadays most Linux users will not install a new Linux system from an iso image.  I was wondering how difficult it is to create a persistent LiveUSB for SpiralLinux?  If someone can show me how to do it, I will try to find time.  Thanks.

**Comment by geckolinux** [comment]
Hi there,

> Nowadays most Linux users will not install a new Linux system from an iso image

In my experience almost all users prefer to just test the system with a live USB and then install it if it works well. Actual long-term users of a live USB are extremely rare. And the other popular method of doing a quick test or review of a distro is with a VM, usually VirtualBox.

However, It should be fairly straightforward to create a persistence partition:
- https://live-team.pages.debian.net/live-manual/html/live-manual/customizing-run-time-behaviours.en.html#556
- https://live-team.pages.debian.net/live-manual/html/live-manual/the-basics.en.html#using-usb-extra-space

**Comment by ping-wu** [comment]
On Fri, Jun 10, 2022 at 3:26 PM GeckoLinux ***@***.***> wrote:

> Hi there,
>
> Nowadays most Linux users will not install a new Linux system from an iso
> image

Absolutely correct!  I was wrong to think, before I ever tried, that the
SpiralLinux is just a bootable iso image similar to Deepin Linux, which no
new Linux users will ever try. I apologize!  This is a true LiveUSB image
(or as Debian community calls it, a hybrid iso).

I rsync'd the SpiralLinux iso into the first fat32-formated partition on a
USB stick, and was able to create a persistent LiveUSB.  I have not tried
anything else, but after a reboot all the changes have been preserved,
including the wireless settings.

This looks like a very exciting Debian derivative.  Will do more
explorations later.  Thanks a whole bunch!

**Reply by geckolinux** [reply]
Excellent, at any rate I'm glad that this topic of live USB persistence came up, because a few users do find it useful. Thanks for testing and reporting back!

**Reply by Raider727** [reply]
I am afraid I was unable to get persistence to work .
      
      can you use the installer to in stall it to a usb . I could not get the installer to point to the usb .
 and In geko Linux is their not an overlay . 
is that what geko has ?

**Reply by geckolinux** [reply]
@ping-wu Could you please explain a bit more how you got it to work?

**Reply by Raider727** [reply]
see I dought  that these instruction work I see many post on diffrent boards . that say they do not .

how but instruction on an over lay like geko . or better to include one with spiralinux .

" In my experience almost all users prefer to just test the system with a live USB and then install it if it works well. Actual long-term users of a live USB are extremely rare. And the other popular method of doing a quick test or review of a distro is with a VM, usually VirtualBox. "

well why did you included it in geko if this was true . You now have some more experience . how about fixing it ?

**Reply by geckolinux** [reply]
> well why did you included it in geko if this was true . You now have some more experience . how about fixing it ?

GeckoLinux is built with a different build tool that includes automatic persistence.

**Reply by ping-wu** [reply]
> @ping-wu Could you please explain a bit more how you got it to work?

Have had a super heavy work load.  Will try to create a persistent USB next week based on the 0626 version to make sure that everything still works then post our procedure.  This is a great OS and persistence should greatly enhance its development.  Sorry for not being able to respond right away.

**Reply by geckolinux** [reply]
No problem, thanks a lot @ping-wu .

**Comment by ping-wu** [comment, marked as answer]
No problem with 0628 edition!  Creating a persistent LiveUSB can be quite complicated, in the following I am listing the simplest procedure:

1.  Create two partitions on a USB3 stick:
      ```
First partition: 4GB, FAT32-formated, Label (optional) EFI
Second partition: 16GB (can be larger or smaller). ext4-formated, Label: "persistence" (no quotes)
```

2.  Copy the content of the iso into the first partition of the usb stick:
```
sudo mount ***0628_x86-64.iso /mnt
rsync -r --progress /mnt/   /media/$USER/EFI/

```
3.  Making the usb persistent:

3a.  edit the /media/$USER/EFI/boot/grub/grub.cfg file by adding " persistence " (no quotes) after "live" (for testing purposes editing the line suffices)
3b.  create a persistence.conf file in the persistence partition then making the entire system persistent:
```
        sudo chmod 777 /media/$USER/persistence
        echo / union >  /media/$USER/persistence/persistence.conf

```
Remove the USB then insert it into a computer, then boot from the USB.

**Reply by Raider727** [reply]
The problem with this is in step 2  

sudo mount ***0628_x86-64.iso /mnt


in my case  When tried it says 

sudo mount SpiralLinux_LXQt_11.220628_x86-64.iso  /mnt
mount: /mnt: WARNING: source write-protected, mounted read-only




rsync -r --progress /mnt/ /media/$USER/EFI/



hpnew:/home/pirate/Downloads # rsync -r --progress /mnt/ /media/$USER/EFI/
sending incremental file list
rsync: [Receiver] mkdir "/media/pirate/EFI" failed: No such file or directory (2)
rsync error: error in file IO (code 11) at main.c(784) [Receiver=3.2.3]


partition  made and formatted with gparted

**Comment by ping-wu** [comment]
1.  In gparted, did you use "Label File System" to label the "EFI" and "persistence" partitions, respectively?
2.  Then, did you mount the  "EFI" and "persistence" partitions? Easiest way is to use nautilus by clicking each partition.

**Reply by Raider727** [reply]
The partition  are labeled as set out . they are mounted I use spacefm to do it .


hpnew:/run/media/root/EFI # ls -a
.  ..  lost+found  SpiralLinux_LXQt_11.220628_x86-64.iso
hpnew:/run/media/root/EFI # sudo mount ***0628_x86-64.iso /mnt
mount: /mnt: WARNING: source write-protected, mounted read-only.
hpnew:/run/media/root/EFI # rsync -r --progress /mnt/ /media/$USER/EFI/
sending incremental file list
rsync: [Receiver] mkdir "/media/pirate/EFI" failed: No such file or directory (2)
rsync error: error in file IO (code 11) at main.c(784) [Receiver=3.2.3]
hpnew:/run/media/root/EFI # 


 and when you go looking on the internet on the debrain method in the early post this is the problem most people seem to have is that the read only  asking a question here was a last resort . I googled every thing I could think of first . many people seem to have this read only problem .

**Comment by ping-wu** [comment]
This is a very simple procedure, don't think too much about it.

The SpiralLinux iso will of course be mounted as read-only, we have made at least hundreds of persistent Debian or Debian based LiveUSBs, this is normal.

Could you do an "ls" command of the /media/pirate directory to see if the "EFI" and "persistence" partitions on the usb are mounted?  Also this is a dumb question, is your first partition (which we conveniently labeled as EFI) is FAT32 formatted?

I am sure you're a very experienced Linux user, please pardon me for asking dumb questions.  But sometimes too much experience can be a hindrance/

**Comment by Raider727** [comment]
pnew:/run/media/root/EFI # ls
SpiralLinux_LXQt_11.220628_x86-64.iso
hpnew:/run/media/root/EFI # sudo mount ***0628_x86-64.iso /mnt
mount: /mnt: WARNING: source write-protected, mounted read-only.
hpnew:/run/media/root/EFI # rsync -r --progress /mnt/ /media/$USER/EFI/
sending incremental file list
rsync: [Receiver] mkdir "/media/pirate/EFI" failed: No such file or directory (2)
rsync error: error in file IO (code 11) at main.c(784) [Receiver=3.2.3]
hpnew:/run/media/root/EFI # 

their is no  /media/pirate directory 

pirate is my user name off the home directory 

EFI is formatted fat 32


their are no dumb question .

**Reply by ping-wu** [reply]
My instruction on creating a persistent SpiralLinux LiveUSB only applies to installed Debian or Debian-derived systems.

**Comment by Raider727** [comment]
So we have played this out and it has reached it 's natural out come
            
            Spiral linux was started to replace geko leap because  of the possibility that suse might end leap
            
            I am working from open suse leap . and using spiralinux as intended . as a possible replacement 
            
            But it is as I said at the start this dose not work .
            
            an no were in the instruction dose it say it will not work from openususe
            
            This has waist a great deal of your time and my . But thanks . may be other will not go down the same road
